datafusion/core/src/datasource/listing/url.rs

```diff
@@ -187,7 +187,10 @@ impl ListingTableUrl {
 
     /// Returns `true` if `path` refers to a collection of objects
     pub fn is_collection(&self) -> bool {
-        self.url.as_str().ends_with(DELIMITER)
+        let mut url = self.url.clone();
+        url.set_query(None);
+        url.set_fragment(None);
+        url.as_str().ends_with(DELIMITER)
```

Comment:
```suggestion
        url.path().ends_with(DELIMITER)
```

I tested that this should work, and it avoids a clone of the URL:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c574fe5ec9fc2a45e33667a2907867f6